pkg/config/config.go

```diff
@@ -130,22 +134,48 @@ func (c *ConfigInfo) writeToFile() error {
 }
 
 // SetConfiguration modifies Odo configurations in the config file
-// as of now only being used for updatenotification
-func (c *ConfigInfo) SetConfiguration(parameter string, value bool) error {
+// as of now being used for timeout, updatenotification
+func (c *ConfigInfo) SetConfiguration(parameter string, value interface{}) error {
 	switch parameter {
+
+	case "timeout":
+		typedval, ok := value.(int)
+		if !ok {
+			return errors.Errorf("type assertion error: unable to set %s", parameter)
+		}
+		if typedval < 0 {
+			return errors.Errorf("cannot set timeout to less than 0")
+		}
+		c.OdoSettings.Timeout = typedval
+
 	case "updatenotification":
-		c.OdoSettings.UpdateNotification = &value
+		typedval, ok := value.(bool)
+		if !ok {
+			return errors.Errorf("type assertion error: unable to set %s", parameter)
+		}
+		c.OdoSettings.UpdateNotification = &typedval
+
 	default:
 		return errors.Errorf("unknown parameter :'%s' is not a parameter in odo config", parameter)
 	}
+
 	err := c.writeToFile()
 	if err != nil {
 		return errors.Wrapf(err, "unable to set %s", parameter)
 	}
 	return nil
 }
 
-// GetupdateNotification returns the value of UpdateNotification from config
+// GetTimeout returns the value of Timeout from config
+func (c *ConfigInfo) GetTimeout() int {
+	// default timeout value is 1
+	if c.OdoSettings.Timeout == 0 {
+		return DefaultTimeout
```

Comment:
why is it returning 1 when the user sets the timeout to 0?
```
[mrinaldas@localhost odo]$ ./odo utils config set timeout 0
[mrinaldas@localhost odo]$ ./odo utils config view
PARAMETER             CURRENT_VALUE
UpdateNotification    true
Timeout               1
```

Reply:
this how the value has been tackled for in the case of parameter not present in `.kube/odo` file

Reply:
but now since we have 0 for no timeout i should find a better way to do same

Reply:
Changed `OdoSettings.Timeout` to int pointer and now it will distinguish 0, nill and all non zeros.
@mik-dass Try now it will work as expected

Reply:
:+1: